core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -130,9 +132,13 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
         .createMapOutputWriter(shuffleId, mapId, numPartitions);
     try {
       if (!records.hasNext()) {
-        partitionLengths = mapOutputWriter.commitAllPartitions();
-        mapStatus = MapStatus$.MODULE$.apply(
-          blockManager.shuffleServerId(), partitionLengths, mapId);
+        mapOutputCommitMessage = mapOutputWriter.commitAllPartitions();
+        taskResult = new MapTaskResult(
```

Comment:
As these lines are repeating you could extract them into a new `def`, like:

```scala
  protected void setTaskResult(MapOutputCommitMessage mapOutputCommitMessage) {
    taskResult = new MapTaskResult(
        MapStatus$.MODULE$.apply(
            blockManager.shuffleServerId(),
            mapOutputCommitMessage.getPartitionLengths(),
            mapId),
        OptionConverters.toScala(mapOutputCommitMessage.getMapOutputMetadata()));
  }
```

With the help of this new `def` and Mockito's spy you can even get rid of the  storing the `mapOutputCommitMessage` for testing purposes only but it has a price (this class cannot be final) for details you can check:
https://github.com/attilapiros/spark/commit/f4578a31beccca58e8b55120847db014a30159d6

Reply:
Ack - didn't address this in my latest patch but will get around to this